core/data/event.go

```diff
@@ -328,25 +299,26 @@ func eventHandler(w http.ResponseWriter, r *http.Request) {
 
 		// Update the fields
 		if from.Device != "" {
-			deviceFound := true
-			d, err := mdc.Device(from.Device)
+			d, err := mdc.CheckForDevice(from.Device)
```

Comment:
As before, only perform this check if MetaDataCheck is enabled. Otherwise, remove MetaDataCheck

Reply:
making changes